pkg/cosign/remote.go

```diff
@@ -95,29 +105,51 @@ func findDuplicate(ctx context.Context, sigImage v1.Image, payload []byte, dupeD
 	return nil, nil
 }
 
-func Upload(ctx context.Context, signature, payload []byte, dst name.Reference, cert, chain string, dupeDetector signature.Verifier) (uploadedSig []byte, err error) {
+type Bundle struct {
+	*models.LogEntryAnon
```

Comment:
(See my comment on the spec above too), we should think through this format a bit since it will become the public API. We may want to version it (though we can do that via the annotation), but we definitely want to explicitly place the fields to separate this type from the Rekor API. Nesting this portion down a bit (Bundle.Verification) might also help place the future keyless part.

Reply:
That makes sense! Maybe it would be better to store the payload, signature and SET in the bundle, and then we can verify those more easily (right now it actually generates the payload during verification instead)

Reply:
+1